### ISSUE TITLE
`await-async-query` does not report inside waitFor

### ISSUE THREAD

**Issue by zaicevas** [testing-library/eslint-plugin-testing-library#396]
```
await waitFor(() => screen.findByText('Click me'))
```

I would expect `await-async-query` to report such usage ⬆️ As I understand it, there is little sense in mixing `waitFor` and `find*`. You either use on or the another 🤷

**Comment by gndelia** [comment]
Hi @zaicevas ! Thanks for posting. The rule you mentioned seems to be related to handling the promises from findBy* methods [as the docs state](https://github.com/testing-library/eslint-plugin-testing-library/blob/main/docs/rules/await-async-query.md)

> Ensure that promises returned by async queries are handled properly.

In that sense, I don't think it should worry about mixing those methods with `waitFor`. I do agree with you that the scenario you pointed out makes little sense. Perhaps it could be an improvement for [prefer-find-by](https://github.com/testing-library/eslint-plugin-testing-library/blob/main/docs/rules/prefer-find-by.md) - what do you think?

**Comment by Belco90** [comment]
I agree with @gndelia: this makes more sense withing `prefer-find-by` rule, since from a point of view of handling promises your example is fine for `await-async-query`.

**Comment by zaicevas** [comment]
Should we reopen this issue?

**Comment by Belco90** [comment]
Sure! Let's reopen it. Are you interested in implementing this improvement?

**Comment by zaicevas** [comment]
> Sure! Let's reopen it. Are you interested in implementing this improvement?

Ahh, missed your comment 😶 I might take care of in an unforeseeable future. In the mean time, if anyone else wants to work on it - please do so :)

**Comment by zaicevas** [comment]
This should also be reported by `prefer-find-by`:
```
      await waitFor(() => expect(screen.getByText('I like carrots.')))
```
